app/presenters/publishing_api/role_presenter.rb

```diff
@@ -58,7 +58,7 @@ def details
         attends_cabinet_type: item.attends_cabinet_type&.name,
         role_payment_type: item.role_payment_type&.name,
         supports_historical_accounts: item.supports_historical_accounts,
-      }
+      }.merge!( item.is_a?(MinisterialRole) ? {seniority: item.seniority} : {} )
```

Comment:
I'd suggest splitting the logic here into a method which might make it easier to read. Perhaps something like this:

```suggestion
      }.merge(ministerial_role_details)
    end
    
    def ministerial_role_details
      return {} unless item.is_a?(MinisterialRole)
      { seniority: item.seniority }
```

Reply:
This was my initial thinking, I was very unsure about the way I had done it but couldn't immediately think of a tidy alternative. Thanks!